### PR TITLE
Module-3: Asynchronous-ASP.NET-Core-APIs

### PR DESCRIPTION
**Pull request by DzmitrySilvanovich**

### CODE REVIEW

File: Ticketing/Ticketing.BAL/Contracts/IVenueService.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+using System.Threading.Tasks;
+using Ticketing.BAL.Model;
+using Ticketing.DAL.Domain;
+
+namespace Ticketing.BAL.Contracts
+{
+    public interface IVenueService
+    {
+        public Task<IEnumerable<VenueReturnModel>> GetVenuesAsync();
+        public Task<IEnumerable<SectionReturnModel>> GetSectionsOfVenue(int venue_id);
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Contracts/IVenueService.cs]: naming: venueId

File: Ticketing/Ticketing.BAL/Enums/Statuses.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Ticketing.BAL.Enums
+{
+    public class Statuses
+    {
+        public enum PaymentStatusEnum { No_payment = 1, Part_payment = 2, Full_payment = 3, Payment_Failed = 4 };
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Enums/Statuses.cs]: naming everywhere. You can refer MSDN: https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names

File: Ticketing/Ticketing.BAL/Services/EventService.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                            IRepository<ShoppingCart> repositoryShoppingCart)
+        {
+            _repositoryEvent = repositoryEvent;
+            _repositorySection = repositorySection;
+            _repositorySeat = repositorySeat;
+            _repositorySeatStatus = repositorySeatStatus;
+            _repositoryPriceType = repositoryPriceType;
+            _repositoryShoppingCart = repositoryShoppingCart;
+        }
+
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Services/EventService.cs:38]: extra lines

File: Ticketing/Ticketing.BAL/Services/EventService.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+
+
+        public async Task<IEnumerable<EventReturnModel>> GetEventsAsync()
+        {
+            var events = await _repositoryEvent.GetAllAsync();
+            return events.Select(e => new EventReturnModel { Id = e.Id, Name = e.Name, EventDate = e.EventDate }).ToList();
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Services/EventService.cs]: mappster nuget + adapt call. https://github.com/MapsterMapper/Mapster

File: Ticketing/Ticketing.BAL/Services/EventService.cs [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        }
+
+        public async Task<List<SeatReturnModel>> GetSeatsAsync(int eventId, int sectionId)
+        {
+            var seatStatuses = (await _repositorySeatStatus.GetAllAsync()).ToList();
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Services/EventService.cs]: why do you need tolist here?

File: Ticketing/Ticketing.BAL/Services/EventService.cs [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+            var priceTypes = (await _repositoryPriceType.GetAllAsync()).ToList();
+
+            var shoppingCarts = await _repositoryShoppingCart.GetAllAsync();
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Services/EventService.cs:43]: filtering on DB side

File: Ticketing/Ticketing.BAL/Services/EventService.cs [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+            var seatIdShippingCart = shoppingCartsFiltered.Select(sh => sh.Id);
+
+            var seats = await _repositorySeat.GetAllAsync();
+            var seatFiltered = seats.Where(s => s.SectionId == sectionId).ToList();
+
+            List<SeatReturnModel> seatReturnModels = seatFiltered.Where(s => seatIdShippingCart.Contains(s.Id))
+                .Select(s => new SeatReturnModel
+                {
+                    SeatId = s.Id,
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Services/EventService.cs]: mappster

File: Ticketing/Ticketing.BAL/Services/EventService.cs [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+                    SectionId = s.SectionId,
+                    RowNumber = s.RowNumber,
+                    SeatNumber = s.SeatNumber,
+                    SeatStatusId = s.SeatStatusId,
+                    NameSeatStatus = string.Empty,
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Services/EventService.cs]: default prop values

File: Ticketing/Ticketing.BAL/Services/EventService.cs [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+                    PriceTypeId = 0,
+                    NamePriceType = string.Empty
+                }).ToList();
+
+            foreach (var item in seatReturnModels)
+            {
+              var priceTypeId = shoppingCartsFiltered.FirstOrDefault(sh => sh.SeatId == item.SeatId)!.PriceTypeId;
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Services/EventService.cs]: offset is broken

File: Ticketing/Ticketing.BAL/Services/PaymentService.cs [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+        public async Task CompletePaymentAsync(int paymentId)
+        {
+            var payment = await _repositoryPayment.GetByIdAsync(paymentId);
+
+            if (payment is null)
+            {
+                return;
+            }
+
+            payment.PaymentStatusId = (int)PaymentStatusEnum.Full_payment;
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Services/PaymentService.cs]: EF is able to translate enum to int, so please use enum in props

File: Ticketing/Ticketing.BAL/Services/PaymentService.cs [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+            await _repositoryPayment.UpdateAsync(payment);
+
+            var shoppingCarts = await _repositoryShoppingCart.GetAllAsync();
+
+            var shoppingCartItems = shoppingCarts.Where(c => c.CartId == payment.CartId).ToList();
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Services/PaymentService.cs:65]: filtering and mapping in db via IQueryble

File: Ticketing/Ticketing.BAL/Services/VenueService.cs [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+        public VenueService(IRepository<Venue> repository, IRepository<Section> repositorySection)
+        {
+            _repository = repository;
+            _repositorySection = repositorySection;
+        }
+
+        public async Task<IEnumerable<VenueReturnModel>> GetVenuesAsync()
+        {
+            var venues = await _repository.GetAllAsync();
+            return venues.Select(v => new VenueReturnModel { Id = v.Id, Name = v.Name }).ToList();
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Services/VenueService.cs]: mappster

File: Ticketing/Ticketing.BAL/Ticketing.BAL.csproj [hunk 13]
```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net7.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Ticketing.BAL.csproj:6]: static analyzer + treat warning as errors

File: Ticketing/Ticketing.DAL/Contracts/IRepository.cs [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+using System.Threading.Tasks;
+using Ticketing.DAL.Domain;
+
+namespace Ticketing.DAL.Contracts
+{
+    public interface IRepository<T>
+    {
+        public Task<T> CreateAsync(T entity);
+
+        public Task<IEnumerable<T>> GetAllAsync();
```
Comment (D9bolic) [inline Ticketing/Ticketing.DAL/Contracts/IRepository.cs]: IQueryble neither IEnumerable

File: Ticketing/Ticketing.DAL/Domains/Order.cs [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Ticketing.DAL.Domain
+{
+    public class Order
+    {
+        public int Id { get; set; }
+        public string? Name { get; set; }
```
Comment (D9bolic) [inline Ticketing/Ticketing.DAL/Domains/Order.cs]: can order name be null?

File: Ticketing/Ticketing.DAL/Domains/Payment.cs [hunk 16]
```diff
@@ -0,0 +1,18 @@
...
+using System.Threading.Tasks;
+
+namespace Ticketing.DAL.Domain
+{
+    public class Payment
+    {
+        public int Id { get; set; }
+        public decimal Amount { get; set; }
+        public int PaymentStatusId { get; set; }
+        public PaymentStatus? PaymentStatus { get; set; }
```
Comment (D9bolic) [inline Ticketing/Ticketing.DAL/Domains/Payment.cs]: can payment status be undifined? also looks like duplication between PaymentStatusId  and PaymentStatus

File: Ticketing/Ticketing.DAL/Domains/Seat.cs [hunk 17]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace Ticketing.DAL.Domain
+{
+    public class Seat
+    {
+        public int Id { get; set; }
+        public int RowNumber { get; set; }
+        public int SeatNumber { get; set; }
+        public int SeatStatusId { get; set; }
+        public SeatStatus? SeatStatus { get; set; }
```
Comment (D9bolic) [inline Ticketing/Ticketing.DAL/Domains/Seat.cs]: can seat status be undefined?

File: Ticketing/Ticketing.DAL/Repositories/ApplicationContext.cs [hunk 18]
```diff
@@ -0,0 +1,93 @@
...
+        public DbSet<Payment> Payments { get; set; } = null!;
+        public DbSet<PaymentStatus> PaymentStatuses { get; set; } = null!;
+        public DbSet<PriceType> PriceTypes { get; set; } = null!;
+        public DbSet<SeatStatus> SeatStatuses { get; set; } = null!;
+        public DbSet<ShoppingCart> ShoppingCarts { get; set; } = null!;
+
+        public ApplicationContext(DbContextOptions<ApplicationContext> options)
+            : base(options)
+        {
+            // uncoment for create database
```
Comment (D9bolic) [inline Ticketing/Ticketing.DAL/Repositories/ApplicationContext.cs]: coments must not be in code

File: Ticketing/Ticketing.DAL/Repositories/CartRepository.cs [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+using Ticketing.DAL.Contracts;
+using Ticketing.DAL.Domain;
+
+namespace Ticketing.DAL.Repositories
+{
+    public class CartRepository : IRepository<Cart>
```
Comment (D9bolic) [inline Ticketing/Ticketing.DAL/Repositories/CartRepository.cs]: do we really need non generic implementations? Please substitute with one generic implementation instead

File: Ticketing/Ticketing.DAL/Ticketing.DAL.csproj [hunk 20]
```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net7.0</TargetFramework>
```
Comment (D9bolic) [inline Ticketing/Ticketing.DAL/Ticketing.DAL.csproj:4]: static analyzers

File: Ticketing/Ticketing.UI/Controlers/EventsController.cs [hunk 21]
```diff
@@ -0,0 +1,47 @@
...
+        {
+            var events = await _eventService.GetEventsAsync();
+
+            foreach (var @event in events)
+            {
+                yield return @event;
+            }
+        }
+
+        //GET /events/{event_id}/sections/{section_id}/ seats
```
Comment (D9bolic) [inline Ticketing/Ticketing.UI/Controlers/EventsController.cs]: why do you need this comments?

File: Ticketing/Ticketing.UI/Controlers/EventsController.cs [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+
+        //GET /events
+        [HttpGet]
+        public async IAsyncEnumerable<EventReturnModel> Get()
+        {
+            var events = await _eventService.GetEventsAsync();
+
+            foreach (var @event in events)
+            {
+                yield return @event;
```
Comment (D9bolic) [inline Ticketing/Ticketing.UI/Controlers/EventsController.cs]: do you really need this yeld, and how you will return like example 204 status code?

File: Ticketing/Ticketing.UI/Controlers/OrdersController.cs [hunk 23]
```diff
@@ -0,0 +1,62 @@
...
+        // POST orders/carts/{cart_id}
+        [HttpPost("{id}")]
+         public async Task<CartStateReturnModel> Post(Guid Id, [FromBody] OrderCartModel orderCartModel)
+        {
+            var result = await _cartService.AddSeatToCartAsync(Id, orderCartModel);
+            return result;
+        }
+
+        // PUT api/<ValuesController>/5
+        [HttpPut("{cart_id}/book")]
```
Comment (D9bolic) [inline Ticketing/Ticketing.UI/Controlers/OrdersController.cs]: naming

Comment (D9bolic) [reply Ticketing/Ticketing.UI/Controlers/OrdersController.cs]: Also with current XML doc how swagger will prepare correct OpenAPI scheme?

File: Ticketing/Ticketing.UI/Controlers/VenuesController.cs [hunk 24]
```diff
@@ -0,0 +1,45 @@
...
+    [ApiController]
+    public class VenuesController : ControllerBase
+    {
+        private IVenueService _venueService;
+
+        public VenuesController(IVenueService venueService)
+        {
+            _venueService = venueService;
+        }
+
```
Comment (D9bolic) [inline Ticketing/Ticketing.UI/Controlers/VenuesController.cs:20]: extra line

File: Ticketing/Ticketing.BAL/Contracts/IVenueService.cs (hunk 1 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Contracts/IVenueService.cs]: done

File: Ticketing/Ticketing.BAL/Enums/Statuses.cs (hunk 2 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Enums/Statuses.cs]: done

File: Ticketing/Ticketing.BAL/Services/EventService.cs (hunk 3 above)
Comment (DzmitrySilvanovich) [inline Ticketing/Ticketing.BAL/Services/EventService.cs:31]: done

File: Ticketing/Ticketing.BAL/Services/EventService.cs (hunk 4 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Services/EventService.cs]: done

File: Ticketing/Ticketing.BAL/Services/EventService.cs (hunk 5 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Services/EventService.cs]: done

File: Ticketing/Ticketing.BAL/Services/EventService.cs (hunk 6 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Services/EventService.cs:43]: done

File: Ticketing/Ticketing.BAL/Services/EventService.cs (hunk 7 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Services/EventService.cs]: done

File: Ticketing/Ticketing.BAL/Services/EventService.cs (hunk 8 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Services/EventService.cs]: done

File: Ticketing/Ticketing.BAL/Services/EventService.cs (hunk 9 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Services/EventService.cs]: done

File: Ticketing/Ticketing.BAL/Services/PaymentService.cs (hunk 10 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Services/PaymentService.cs]: done

File: Ticketing/Ticketing.BAL/Services/PaymentService.cs (hunk 11 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Services/PaymentService.cs:65]: done

File: Ticketing/Ticketing.BAL/Services/VenueService.cs (hunk 12 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Services/VenueService.cs]: done

File: Ticketing/Ticketing.BAL/Ticketing.BAL.csproj (hunk 13 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Ticketing.BAL.csproj:6]: done

File: Ticketing/Ticketing.DAL/Contracts/IRepository.cs (hunk 14 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.DAL/Contracts/IRepository.cs]: done

File: Ticketing/Ticketing.DAL/Domains/Order.cs (hunk 15 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.DAL/Domains/Order.cs]: done

File: Ticketing/Ticketing.DAL/Domains/Payment.cs (hunk 16 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.DAL/Domains/Payment.cs]: done

File: Ticketing/Ticketing.DAL/Domains/Seat.cs (hunk 17 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.DAL/Domains/Seat.cs]: done

File: Ticketing/Ticketing.DAL/Repositories/ApplicationContext.cs (hunk 18 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.DAL/Repositories/ApplicationContext.cs]: done

File: Ticketing/Ticketing.DAL/Repositories/CartRepository.cs (hunk 19 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.DAL/Repositories/CartRepository.cs]: done

File: Ticketing/Ticketing.DAL/Ticketing.DAL.csproj (hunk 20 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.DAL/Ticketing.DAL.csproj:4]: done

File: Ticketing/Ticketing.UI/Controlers/EventsController.cs (hunk 21 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.UI/Controlers/EventsController.cs]: done

File: Ticketing/Ticketing.UI/Controlers/EventsController.cs (hunk 22 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.UI/Controlers/EventsController.cs]: done

File: Ticketing/Ticketing.UI/Controlers/OrdersController.cs (hunk 23 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.UI/Controlers/OrdersController.cs]: done

File: Ticketing/Ticketing.UI/Controlers/VenuesController.cs (hunk 24 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.UI/Controlers/VenuesController.cs:20]: done

File: Ticketing/Ticketing.BAL/Contracts/IPaymentService.cs [hunk 25]
```diff
@@ -0,0 +1,18 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Ticketing.BAL.Model;
+
+namespace Ticketing.BAL.Contracts
+{
+    public interface IPaymentService
+    {
+        Task<PaymentStatusReturnModel> GetPaymentSatatusAsync(int paymentId);
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Contracts/IPaymentService.cs]: misspelling

File: Ticketing/Ticketing.BAL/Contracts/IPaymentService.cs [hunk 26]
```diff
@@ -0,0 +1,18 @@
...
+
+        Task CompletePaymentAsync(int paymentId);
+
+        Task FailedPaymentAsync(int paymentId);
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Contracts/IPaymentService.cs]: Shall be action, now statement. FailPaymentAsync or smth like that

File: Ticketing/Ticketing.BAL/Services/CartService.cs [hunk 27]
```diff
@@ -0,0 +1,116 @@
...
+        public CartService(Repository<ShoppingCart> repository, Repository<Seat> repositorySeat, Repository<Payment> repositoryPayment)
+        {
+            _repositoryShoppingCart = repository;
+            _repositorySeat = repositorySeat;
+            _repositoryPayment = repositoryPayment;
+        }
+
+        public async Task<CartStateReturnModel> AddSeatToCartAsync(Guid CartId, OrderCartModel orderCartModel)
+        {
+
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Services/CartService.cs]: extra line

File: Ticketing/Ticketing.BAL/Services/CartService.cs [hunk 28]
```diff
@@ -0,0 +1,116 @@
...
+            {
+                await _repositoryShoppingCart.CreateAsync(shoppingCartDto);
+            }
+            else
+            {
+                item.PriceTypeId = orderCartModel.PriceTypeId;
+                item.Price = orderCartModel.Price;
+            }
+
+            await _repositoryShoppingCart.UpdateAsync(shoppingCartDto);
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Services/CartService.cs]: can be moved to the else, as nothing was changed after the creation. Also looks like private method GetOrCreate

File: Ticketing/Ticketing.BAL/Services/CartService.cs [hunk 29]
```diff
@@ -0,0 +1,116 @@
...
+                CartId = cartId,
+                TotalAmount = totalAmount
+            };
+        }
+
+        public async Task<int> BookSeatToCartAsync(Guid cartId)
+        {
+            var shoppingCarts = await _repositoryShoppingCart.GetAllAsync();
+
+            var shoppingCartItems = shoppingCarts.Where(c => c.CartId == cartId).ToList();
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Services/CartService.cs:70]: Please move to the one LINQ statement all 3

File: Ticketing/Ticketing.BAL/Services/CartService.cs [hunk 30]
```diff
@@ -0,0 +1,116 @@
...
+
+            var shoppingCartSeats = shoppingCartItems.Select(sh => sh.SeatId).ToList();
+
+            var allSeats = await _repositorySeat.GetAllAsync();
+
+            var seats = allSeats.Where(s => shoppingCartSeats.Contains(s.Id));
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Services/CartService.cs:76]: the same

File: Ticketing/Ticketing.BAL/Services/CartService.cs [hunk 31]
```diff
@@ -0,0 +1,116 @@
...
+            };
+
+            var newPayment = await _repositoryPayment.CreateAsync(payment);
+
+            return newPayment.Id;
+        }
+
+        public async Task DeleteSeatForCartAsync(Guid cartId, int eventId, int seatId)
+        {
+            var shoppingCarts = await _repositoryShoppingCart.GetAllAsync();
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Services/CartService.cs:100]: to the one LINQ please. Also ToListAsync for all places like that

File: Ticketing/Ticketing.BAL/Services/CartService.cs [hunk 32]
```diff
@@ -0,0 +1,116 @@
...
+
+            foreach (var item in shoppingCartItems)
+            {
+                await _repositoryShoppingCart.DeleteAync(item);
+            }
+        }
+
+        public async Task<IEnumerable<ShoppingCartReturnModel>> CartItemsAsync(Guid cartId)
+        {
+             var items = await _repositoryShoppingCart.GetAllAsync();
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Services/CartService.cs:112]: styling

File: Ticketing/Ticketing.BAL/Services/CartService.cs [hunk 33]
```diff
@@ -0,0 +1,116 @@
...
+            return items.Where(i => i.CartId == cartId).AsQueryable().ProjectToType<ShoppingCartReturnModel>().ToList();
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Services/CartService.cs]: why do you have explicit AsQueryable here?

File: Ticketing/Ticketing.BAL/Services/EventService.cs [hunk 34]
```diff
@@ -0,0 +1,68 @@
...
+                          {
+                              SeatId = seat.Id,
+                              seat.SectionId,
+                              seat.RowNumber,
+                              seat.SeatNumber,
+                              seat.SeatStatusId,
+                              NameSeatStatus = seatStatus.Name,
+                              shoppingCart.PriceTypeId,
+                              NamePriceType = priceType.Name
+                          }).AsQueryable().ProjectToType<SeatReturnModel>().ToList();
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Services/EventService.cs]: ToListASync + IQuerible as a repository contract, to not do AsQueryable

File: Ticketing/Ticketing.BAL/Services/PaymentService.cs [hunk 35]
```diff
@@ -0,0 +1,109 @@
...
+            {
+                return;
+            }
+
+            payment.PaymentStatusId = PaymentState.PaymentFailed;
+            await _repositoryPayment.UpdateAsync(payment);
+
+            var shoppingCarts = await _repositoryShoppingCart.GetAllAsync();
+
+            var shoppingCartItems = shoppingCarts.Where(c => c.CartId == payment.CartId).ToList();
```
Comment (D9bolic) [inline Ticketing/Ticketing.BAL/Services/PaymentService.cs:94]: same for all other places

File: Ticketing/Ticketing.BAL/Contracts/IPaymentService.cs (hunk 25 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Contracts/IPaymentService.cs]: done

File: Ticketing/Ticketing.BAL/Contracts/IPaymentService.cs (hunk 26 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Contracts/IPaymentService.cs]: done

File: Ticketing/Ticketing.BAL/Services/CartService.cs (hunk 27 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Services/CartService.cs]: done

File: Ticketing/Ticketing.BAL/Services/CartService.cs (hunk 28 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Services/CartService.cs]: done

File: Ticketing/Ticketing.BAL/Services/CartService.cs (hunk 29 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Services/CartService.cs:70]: it doesn't work that way

File: Ticketing/Ticketing.BAL/Services/CartService.cs (hunk 30 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Services/CartService.cs:76]: it doesn't work that way

File: Ticketing/Ticketing.BAL/Services/CartService.cs (hunk 31 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Services/CartService.cs:100]: it doesn't work that way

File: Ticketing/Ticketing.BAL/Services/CartService.cs (hunk 32 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Services/CartService.cs:112]: done

File: Ticketing/Ticketing.BAL/Services/CartService.cs (hunk 33 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Services/CartService.cs]: done

File: Ticketing/Ticketing.BAL/Services/EventService.cs (hunk 34 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Services/EventService.cs]: done

File: Ticketing/Ticketing.BAL/Services/PaymentService.cs (hunk 35 above)
Comment (DzmitrySilvanovich) [reply Ticketing/Ticketing.BAL/Services/PaymentService.cs:94]: it doesn't work that way

**Pull request merged**